File: LC-Python-Solution/219. Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:

        nMap = {}
        for i,num in enumerate(nums):

            if ((num in nMap)and(abs(i-nMap[num])<=k)):
                return True

            nMap[num] = i
        
        return False





        

File: LC_JS-SOLUTIONS/219. Contains Duplicate II.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function (nums, k) {
  nMap = {};

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] in nMap && Math.abs(i - nMap[nums[i]]) <= k) {
      return true;
    }

    nMap[nums[i]] = i;
  }

  return false;
};
